test/e2e-go/features/participation/onlineOfflineParticipation_test.go

```diff
@@ -159,35 +174,45 @@ func TestNewAccountCanGoOnlineAndParticipate(t *testing.T) {
 	newAccountStatus, err := client.AccountInformation(newAccount)
 	a.NoError(err, "client should be able to get information about new account")
 	a.Equal(basics.Online.String(), newAccountStatus.Status, "new account should be online")
+
+	nodeStatus, _ = client.Status()
+	fundedRoundBefore := nodeStatus.LastRound
+	
 	// account receives almost all of rich account's stake (minus enough to
 	// keep it over MinBalance), so it will be selected for participation
 	amountToSend := richBalance - 3*transactionFee - amountToSendInitial - minAcctBalance
 	fixture.SendMoneyAndWait(onlineRound, amountToSend, transactionFee, richAccount, newAccount)
 
 	nodeStatus, _ = client.Status()
-	fundedRound := nodeStatus.LastRound
+	fundedRoundAfter := nodeStatus.LastRound	
 
-	params, err := client.ConsensusParams(nodeStatus.LastRound)
+	params, err := client.ConsensusParams(fundedRoundAfter)
 	a.NoError(err)
-	lookbackRound := balanceRound(basics.Round(nodeStatus.LastRound), params)
-	delta := int64(nodeStatus.LastRound) - int64(lookbackRound)
-	a.True(delta > 0)
+	fundedAccountRoundBefore := balanceRoundOf(basics.Round(fundedRoundBefore), params)
+	fundedAccountRoundAfter := balanceRoundOf(basics.Round(fundedRoundAfter), params)
 
 	// Need to wait for funding to take effect on selection, then we can see if we're participating
 	// Stop before the account should become eligible for selection so we can ensure it wasn't
-	err = fixture.WaitForRoundWithTimeout(fundedRound + uint64(delta) - 1)
+	err = fixture.ClientWaitForRound(fixture.AlgodClient, uint64(fundedAccountRoundBefore),
+		time.Duration(uint64(globals.MaxTimePerRound) * uint64(fundedAccountRoundBefore)))
+	err = fixture.WaitForRoundWithTimeout(uint64(fundedAccountRoundBefore))
```

Comment:
why fundedAccountRoundBefore and not fundedAccountRoundAfter ?
You want to get to a round at which the newly founded account can vote. For fundedAccountRoundBefore, the account won't be able to vote.

Reply:
This is rewritten with different logic.